### PR TITLE
feat: Infer artist series when creating alert form artwork screen

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [ONYX-566] 

- Related Force PR: https://github.com/artsy/force/pull/13244
- 
### Description

This change adds artist series as an initial criterion when creating the alert from the artwork screen.

| | |
| --- | --- |
|Simulator Screenshot - iPhone 15 Plus - 2023-12-08 at 12 37 19 |Simulator Screenshot - iPhone 15 Plus - 2023-12-08 at 12 36 59|

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Infer artist series when creating alert form artwork screen - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-566]: https://artsyproduct.atlassian.net/browse/ONYX-566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against dcc3003737d7ba6cd676544cd9e9d4cc7a4a9827
</p>

File: src/app/Components/Artist/ArtistArtworks/CreateArtworkAlertModal.tsx [hunk 1]
```diff
@@ -30,6 +31,13 @@ export const CreateArtworkAlertModal: React.FC<CreateArtworkAlertModalProps> = (
   const data = useFragment<CreateArtworkAlertModalProps["artwork"]>(
     graphql`
       fragment CreateArtworkAlertModal_artwork on Artwork {
+        artistSeriesConnection(first: 5) {
```
Comment (olerichter00) [inline src/app/Components/Artist/ArtistArtworks/CreateArtworkAlertModal.tsx:34]: Using 5 as the maximum number of artist series used when creating an alert would cover 99.98% of the cases.(context: https://github.com/artsy/force/pull/13244#discussion_r1419289315)

File: src/app/Scenes/SavedSearchAlert/pillExtractors.ts [hunk 2]
```diff
@@ -150,6 +151,16 @@ export const extractAdditionalGeneIDsPills = (values: string[]): SavedSearchPill
   })
 }
 
+export const extractPillFromArtistSeriesIDs = (values: string[]): SavedSearchPill[] => {
+  return values.map((value) => {
+    return {
+      label: toTitleCase(value.replaceAll("-", " ")),
```
Comment (olerichter00) [inline src/app/Scenes/SavedSearchAlert/pillExtractors.ts:157]: This isn't ideal, but it should be good enough for now. We have already moved the pill/label generation to Metaphysics and will remove this code soon in favor of using the backend-generated labels.

Screenshot 2023-12-08 at 12 30 02

File: src/app/Components/Artist/ArtistArtworks/CreateArtworkAlertModal.tsx [hunk 3]
```diff
@@ -52,38 +61,40 @@ export const CreateArtworkAlertModal: React.FC<CreateArtworkAlertModalProps> = (
...
       currentArtworkID={data.internalID}
     />
   )
 }
 
-export const computeArtworkAlertProps = (
+export const useComputeArtworkAlertProps = (
   artwork: CreateArtworkAlertModal_artwork$data | BrowseSimilarWorks_artwork$data
 ) => {
+  const enableArtistSeriesFilter = useFeatureFlag("AREnableArtistSeriesFilter")
```
Comment (olerichter00) [inline src/app/Components/Artist/ArtistArtworks/CreateArtworkAlertModal.tsx:92]: Still an open discussion whether to use a second feature flag here ([Slack Thread](https://artsy.slack.com/archives/C05EQL4R5N0/p1701965113553309?thread_ts=1701954132.712389&cid=C05EQL4R5N0))

**Review by MounirDhahri — APPROVED** [review]
LGTM! small comments none are blocking

**Comment by olerichter00** [comment]
Closing this PR because we only want to suggest artist series ([Slack Thread](https://artsy.slack.com/archives/C05EQL4R5N0/p1702491201631999))

**Pull request closed without merging**